src/trait_/zeroize_on_drop.rs

```diff
@@ -18,38 +21,40 @@ impl TraitImpl for ZeroizeOnDrop {
 		DeriveTrait::ZeroizeOnDrop { crate_: None }
 	}
 
-	fn parse_derive_trait(&self, list: MetaList) -> Result<DeriveTrait> {
+	fn parse_derive_trait(
+		&self,
+		_span: Span,
+		list: Punctuated<Meta, Token![,]>,
+	) -> Result<DeriveTrait> {
 		// This is already checked in `DeriveTrait::from_stream`.
-		debug_assert!(!list.nested.is_empty());
+		debug_assert!(!list.is_empty());
 
 		let mut crate_ = None;
 
-		for nested_meta in list.nested {
-			match &nested_meta {
-				NestedMeta::Meta(Meta::Path(path)) => {
-					return Err(Error::option_trait(path.span(), self.as_str()))
-				}
-				NestedMeta::Meta(Meta::NameValue(name_value)) => {
+		for meta in list {
+			match &meta {
+				Meta::Path(path) => return Err(Error::option_trait(path.span(), self.as_str())),
+				Meta::NameValue(name_value) => {
 					if name_value.path.is_ident("crate") {
 						// Check for duplicate `crate` option.
 						if crate_.is_none() {
-							if let Lit::Str(lit_str) = &name_value.lit {
-								match lit_str.parse::<Path>() {
-									Ok(path) => {
-										if path == util::path_from_strs(&["zeroize"]) {
-											return Err(Error::path_unnecessary(
-												path.span(),
-												"::zeroize",
-											));
-										}
-
-										crate_ = Some(path);
-									}
+							let path = match &name_value.value {
+								Expr::Lit(ExprLit {
+									lit: Lit::Str(lit_str),
+									..
+								}) => match lit_str.parse::<Path>() {
+									Ok(path) => path,
 									Err(error) => return Err(Error::path(lit_str.span(), error)),
-								}
-							} else {
-								return Err(Error::option_syntax(name_value.lit.span()));
+								},
+								Expr::Path(ExprPath { path, .. }) => path.clone(),
+								_ => return Err(Error::option_syntax(name_value.value.span())),
+							};
+
+							if path == util::path_from_strs(&["zeroize"]) {
+								return Err(Error::path_unnecessary(path.span(), "::zeroize"));
 							}
```

Comment:
Would it make sense to reuse code between this and the `derive_where` `crate`?

Reply:
That's a good idea, I will give it a shot.

Reply:
After a pathetic attempt I didn't really like it. Feel free to give it a shot yourself.

Reply:
I'd be fine with keeping is as is then.